src/Sylius/Bundle/CoreBundle/spec/Factory/OrderFactorySpec.php

```diff
@@ -0,0 +1,72 @@
+<?php
+
+/*
+ * This file is part of the Sylius package.
+ *
+ * (c) Paweł Jędrzejewski
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+declare(strict_types=1);
+
+namespace spec\Sylius\Bundle\CoreBundle\Factory;
+
+use PhpSpec\ObjectBehavior;
+use Sylius\Component\Core\Model\Address;
+use Sylius\Component\Core\Model\Channel;
+use Sylius\Component\Core\Model\ChannelInterface;
+use Sylius\Component\Core\Model\Customer;
+use Sylius\Component\Core\Model\CustomerInterface;
+use Sylius\Component\Core\Model\Order;
+use Sylius\Component\Currency\Model\Currency;
+use Sylius\Component\Resource\Factory\FactoryInterface;
+
+final class OrderFactorySpec extends ObjectBehavior
+{
+    public function let(FactoryInterface $baseFactory): void
+    {
+        $this->beConstructedWith($baseFactory);
+    }
+
+    public function it_creates_an_order(FactoryInterface $baseFactory): void
+    {
+        $someOrder = new Order();
+        $baseFactory->createNew()->willReturn($someOrder);
+
+        $this->createNew()->shouldReturn($someOrder);
+    }
```

Comment:
I would be for mocking an order by its interface in the arguments of the spec function, the same applies to the below scenario